packages/editor/src/browser/editor.ts

```diff
@@ -40,7 +40,8 @@ export interface TextEditor extends Disposable, TextEditorSelection {
     readonly onFocusChanged: Event<boolean>;
 
     revealPosition(position: Position): void;
-    revealRange(range: Range): void;
+    revealRange(range: Range, revealAt?: VerticalRevealType): void;
```

Comment:
Could we use options pattern as at other places? It allows adding new options later, like to specify how a range should be revealed horizontally.
```ts
/**
 * by default a range is revealed vertically at center
 */
revealRange(range: range, options?: RevealRangeOptions): void;

interface RevealRangeOptions {
    at: 'center' | 'top' | 'centerIfOutsideViewport'
}
```
Also i find it is quite readable at the client side:
```ts
editor.reveal(range, { at: 'top' })
```

Reply:
sounds good!

Reply:
I've updated it.